jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.c

```diff
@@ -484,8 +577,13 @@ ecma_builtin_string_prototype_object_replace (ecma_value_t this_value, /**< this
     {
       if (!memcmp (curr_p, search_buf_p, search_size))
       {
-        const lit_utf8_size_t byte_offset = (lit_utf8_size_t) (curr_p - replace_ctx.string_p);
-        replace_ctx.builder = ecma_stringbuilder_create_raw (replace_ctx.string_p, byte_offset);
+        const lit_utf8_size_t byte_offset = (lit_utf8_size_t) (curr_p - last_match_end_p);
```

Comment:
Please rename this to something more appropriate, may be prefix_size or somethign similar.